### PR TITLE
Lighting Overhaul & General process improvements

### PR DESCRIPTION
**Pull request by Lohikar**
This is it. The big one.
**Risk: Very large. This modifies or rewrites several important systems.**
Some things still need balancing, but that's probably better done if/when this hits dev.

changes:
- New smooth lighting system.
- ~~Machinery split into three processes: machinery, powernet, pipenet~~ Removed due to breakage. Refactored into multi-step process.
- Mob process rewritten.
- NanoUI process rewritten.
- Objects process rewritten.
- Tweaked color output of station lights.
- Slime core lights now emit colored light.
- Fixed light update frequency issue with fire alarms, hydroponics trays, and airlocks.
- Increased light emission from bolted airlocks.
- Miscellaneous performance improvements.
- New datum pool implementation.
- New lighting usage profiler.
- Lighting system now tracks UV light, which is not visible to players.
- Space now has a parallax effect.
- Disabled Spin View verbs due to incompatibility with the new lighting system.
- Disabled hallucination view spin due to incompatibility with the new lighting system.
- Lighting system now initializes in the lobby before the round starts to reduce BoR deadtime.
- Added UV light tracking to lighting engine; dionae now gain energy exclusively from UV light.
- Added colored lighting to a few consoles that used default (white) light.

Todo/Issues:
- [x] Fix dionae light checking.
- [x] Verify vamp veilwalk light checking.
- [x] Verify plant light checking.
- [X] Fix flashlight lag.
- [x] Fix absolute darkness not being `rgb(0,0,0)`.
- [x] Fix fire alarms not updating lights on sec level change.
- [x] Fix lighting effects from Nar'sie's arrival and Supermatter Cascades.
- [x] Fix shuttle movement invalidating & pooling lighting tiles.
- [x] Decrease brightness of nightmode so that it actually has a visible effect.
- [x] Make border of darkness<->space look less terrible.
- [x] Fix viewport rotation (such as with Rotate View verbs).
- [x] Rebalance station lighting.
- [ ] Rebalance portable light sources.
- [x] Rebalance APC, Fire Alarm & Air Alarm light emission.
- [x] Rebalance emergency floodlight light emission.
- [ ] Rebalance voidsuit/hardsuit helmet light emission.
- [x] Turning on portable light sources should bypass the light scheduler.
- [x] Toggling lights with an APC should bypass the light scheduler. - Shelved.
- [x] Toggling lights with a lightswitch should bypass the light scheduler. - Shelved.
- [x] Opening doors should trigger a light revalidation that bypasses the light scheduler.
- [x] Parallax does not work on non-instant shuttles, like the emergency shuttle. - Fixed by restoring old appearance - looks bad, sprites need to be darkened to match current space appearance.
- [x] Nar-sie causes runtimes in lighting_overlay.dm -> cannot read from list. - Related to corners. - Drop pods also.
- [x] Mesons & other glasses are fucked.
- [x] BYOND native darkness does not work.
- [x] `see_in_dark` does not work.
- [x] Make parallax toggle not require client restart.
- [x] Investigate potential short-circuit in scheduler - overlay updates happening earlier than they should be.
- [x] Lighting updates should check server load/lag and decide between instant updates and the scheduler. - Shelved.

### CODE REVIEW

**Comment by skull132** [comment]
Mesons and see_in_dark fixed in #1642 .

**Comment by Lohikar** [comment]
DNM again because git undid some stuff at some point and it needs fixing.

File: code/modules/lighting/lighting_profiler.dm [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Writes lighting updates to the database.
+// FOR DEBUGGING ONLY!
+
+var/DBQuery/lprof_q
```
Comment (skull132) [inline code/modules/lighting/lighting_profiler.dm]: This doesn't need to exist as a global variable. Write it as a static one inside lprof_write().

File: code/modules/lighting/lighting_profiler.dm [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+/proc/lprof_init()
+	lprof_q = dbcon.NewQuery({"INSERT INTO ss13dbg_lighting (time,type,name,loc_name,x,y,z) 
+		VALUES (:time,:type,:name,:loc_name,:x,:y,:z);"})
+
+/proc/lprof_write(var/atom/movable/obj, var/type = "UNKNOWN")
+	if (!lighting_profiling || !obj || !dbcon.IsConnected())
+		return
+
```
Comment (skull132) [inline code/modules/lighting/lighting_profiler.dm]:
```
	var/static/DBQuery/lprof_q
	if (!lprof_q)
		lprof_q = dbcon.NewQuery({"INSERT INTO ss13dbg_lighting (time,type,name,loc_name,x,y,z)
			VALUES (:time,:type,:name,:loc_name,:x,:y,:z);"})
```

File: code/modules/lighting/lighting_profiler.dm [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+// FOR DEBUGGING ONLY!
+
+var/DBQuery/lprof_q
+
+/proc/lprof_init()
+	lprof_q = dbcon.NewQuery({"INSERT INTO ss13dbg_lighting (time,type,name,loc_name,x,y,z) 
+		VALUES (:time,:type,:name,:loc_name,:x,:y,:z);"})
+
+/proc/lprof_write(var/atom/movable/obj, var/type = "UNKNOWN")
+	if (!lighting_profiling || !obj || !dbcon.IsConnected())
```
Comment (skull132) [inline code/modules/lighting/lighting_profiler.dm]: Last check should be `!establish_db_connection(dbcon)`. The first thing establish_db_connection does is check IsConnected().

File: html/changelogs/lohikar-lighting.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+author: Lohikar
+delete-after: True
+changes: 
+  - rscadd: "Lighting is now smoothed across tiles instead of being per-tile."
```
Comment (skull132) [inline html/changelogs/lohikar-lighting.yml]: Missing note regarding that this system is an import from VG/Goon.

**Pull request merged**